### DISCUSSION TITLE
Issue with AutoTS Model Execution in PyInstaller Executable

### DISCUSSION THREAD

**Discussion by philbuii** [winedarksea/AutoTS#248] [Q&A]
I’ve built a Python app using the AutoTS library for forecasting. It works fine when run directly with Python, but after converting it to an executable using PyInstaller, I encounter the following error:

`Evaluation Metrics are missing and all models have failed, by an error in template or metrics.
There are many possible causes for this, bad parameters, environment, or an unreported bug.
Usually this means you are missing required packages for the models like fbprophet or gluonts,
or that the models in model_list are inappropriate for your data.
A new starting template may also help. KeyError('smape')
`

Has anyone faced similar issues or know why this might be happening? Are there specific dependencies or settings required to ensure AutoTS works properly with PyInstaller?

Any guidance on resolving this would be greatly appreciated

**Comment by winedarksea** [comment]
This error occurs when something is really broken. 
One possible issue I see here is that you may be requesting a model set from model_list arg, but don't assure that the necessary dependencies are installed with pyinstaller?
You'll need to provide more details about what code you are running for me to really figure it out.

**Reply by philbuii** [reply]
Thanks for your response!

I'm trying to run a small forecasting script where users can load an Excel file and forecast the data, using Tkinter for the GUI.

Here’s a snippet of how I initialize the AutoTS model:
`...
model = AutoTS(
                    forecast_length=12,
                    frequency='infer',
                    prediction_interval=0.9,
                    ensemble='simple',
                    model_list="superfast",
                    transformer_list="superfast",
                    drop_most_recent=1,
                    validation_method='seasonal 168',
                    max_generations=5,
                    num_validations=2,
                    model_interrupt=True,
                    n_jobs='3',
                    )
                    ...`

Notably, this error only occurs when I use the --windowed or --noconsole options, as I didn't want the console to open with the GUI. Without --windowed/--noconsole, the executable runs fine, but with those options, I get the error.

Could this be related to how PyInstaller packages the environment or misses some dynamic imports?

**Reply by winedarksea** [reply]
I am not really sure, as it has been years since I ever tried to make a packaged executable file.
Some things to look at:

- test and make sure scikit learn, pandas, and numpy are all present in the executable. That should be the bare minimum, if those are present it should run 'superfast' just fine
- double check the imported data (maybe print `model.df_wide_numeric` head to log and see if it looks somewhat reasonable)
- try setting n_jobs = 1, parallelization could be the issue inside the executable somehow?
